src/packet_analysis/protocol/iptunnel/IPTunnel.cc

```diff
@@ -183,12 +183,8 @@ std::unique_ptr<Packet> build_inner_packet(Packet* outer_pkt, int* encap_index,
 	uint32_t consumed_len = outer_pkt->cap_len - inner_cap_len;
 	uint32_t inner_wire_len = outer_pkt->len - consumed_len;
 
-	pkt_timeval ts;
-	ts.tv_sec = static_cast<time_t>(run_state::current_timestamp);
-	ts.tv_usec = static_cast<suseconds_t>(
-		(run_state::current_timestamp - static_cast<double>(ts.tv_sec)) * 1000000);
-
-	auto inner_pkt = std::make_unique<Packet>(link_type, &ts, inner_cap_len, inner_wire_len, data);
+	auto inner_pkt = std::make_unique<Packet>(link_type, &outer_pkt->ts, inner_cap_len,
```

Comment:
Does this work correctly for multiply-nested packets, if one of the nests has a weird timestamp? The previous method would always use the timestamp for the outermost frame, but now it'll potentially use one from an inner frame that might not match.

Reply:
I didn't think too much as this aligns with `ProcessEncapsulatedPacket` that do the same if `pkt` is provided. Also, think it's more intuitive to determine the timestamp based on input parameters rather than global state (would make unit testing easier as well).

I'm unclear where we would get a "weird" timestamp from. At least currently, it seems the `ts` field would anyhow be the same as the global network time  for `build_inner_packet()` users.

This borders the scenario where tunnel metadata contains timestamps of inner packets. Though that would certainly require more surgery and you wouldn't want to use the global network time then either.

Going to put something about the timestamp in the docstring, but otherwise resolving.